frontends/web/src/routes/account/walletconnect/components/incoming-pairing/incoming-pairing.tsx

```diff
@@ -0,0 +1,108 @@
+
+/**
+ * Copyright 2023 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useTranslation } from 'react-i18next';
+import { CoreTypes, SignClientTypes } from '@walletconnect/types';
+import { buildApprovedNamespaces, getSdkError } from '@walletconnect/utils';
+import { Button } from '../../../../../components/forms';
+import { alertUser } from '../../../../../components/alert/Alert';
+import { SUPPORTED_CHAINS, web3wallet } from '../../utils';
+import styles from './incoming-pairing.module.css';
+
+type TIncomingPairingProps = {
+    currentProposal: SignClientTypes.EventArguments['session_proposal'];
+    pairingMetadata: CoreTypes.Metadata;
+    receiveAddress: string;
+    onReject: () => void;
+    onApprove: () => void;
+}
+
+const PairingContainer = ({ pairingMetadata }: {pairingMetadata: TIncomingPairingProps['pairingMetadata']}) => {
+  const { name, description, url, icons } = pairingMetadata;
+  const hasIcon = icons && icons.length > 0;
+  return (
+    <div className={styles.pairingDetailContainer}>
+      <div>
+        <p className={styles.metadata}>{name}</p>
+        <p className={styles.metadata}>{description}</p>
+        <p className={styles.metadata}>{url}</p>
+      </div>
+      {hasIcon && <img src={icons[0]} alt="logo" />}
+    </div>
+  );
+};
+
+export const WCIncomingPairing = ({
+  currentProposal,
+  pairingMetadata,
+  receiveAddress,
+  onReject,
+  onApprove
+}: TIncomingPairingProps) => {
+  const { t } = useTranslation();
+  const handleApprovePairing = async () => {
+    try {
+      const { id, params } = currentProposal;
+      const { requiredNamespaces } = params;
+      const accounts: string[] = [];
+      Object.keys(requiredNamespaces).forEach(key => {
+        requiredNamespaces[key].chains?.map(chain => accounts.push(`${chain}:${receiveAddress}`));
+      });
+
+      const namespaces = buildApprovedNamespaces({
```

Comment:
maybe add a comment what this does, or is it just to have valid object structure?

Reply:
This is a util function provided by wallet connect library: https://docs.walletconnect.com/2.0/web3wallet/wallet-usage?platform=web#namespace-builder . 

I'll add a comment for this. Thanks :)